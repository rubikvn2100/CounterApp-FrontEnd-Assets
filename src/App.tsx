import React from "react";
import { BrowserRouter, Route, Routes } from "react-router-dom";
import NavBar from "./components/Navigation/NavBar";

const App: React.FC = () => {
  return (
    <BrowserRouter>
      <NavBar />
      <Routes>
        <Route
          path="/"
          element={
            <div>
              <h1>Dummy Home Page</h1>
            </div>
          }
        />
      </Routes>
    </BrowserRouter>
  );
};

export default App;
