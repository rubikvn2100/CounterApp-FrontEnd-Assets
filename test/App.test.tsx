import React from "react";
import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import App from "../src/App";

describe("App component", () => {
  test("renders Hello World text", () => {
    render(<App />);
    expect(screen.getByText("Hello World")).toBeDefined();
  });

  test("matches the snapshot", () => {
    const tree = renderer
      .create(<App />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
