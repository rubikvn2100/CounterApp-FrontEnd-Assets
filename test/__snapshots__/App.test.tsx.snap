// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component matches the snapshot 1`] = `
[
  <nav>
    <div
      className="undefined undefined"
    >
      <div>
        <span>
          Counter
        </span>
        <span>
          App
        </span>
      </div>
      <div
        onClick={[Function]}
      >
        â˜°
      </div>
      <div>
        <a
          className=""
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </div>
    </div>
  </nav>,
  <div>
    <div>
      <div>
        <div>
          Counter: 
          0
        </div>
        <div>
          Connecting
        </div>
      </div>
    </div>
  </div>,
]
`;
